### PR TITLE
Ugliest code of all time

### PR DESCRIPTION
**Pull request by lekko-jonathan**

### CODE REVIEW

**Comment by lekko-jonathan** [comment]
This is not an actual pull request, just the best way to share it with Sergey

**Comment by lekko-jonathan** [comment]
For the included default.ts it prints out:
```
syntax = "proto3";

message Dog_HealthRecord_Vaccinations {
  string type = 1;
  int32 date = 2;
}

message Dog_HealthRecord {
  repeated Dog_HealthRecord_Vaccinations vaccinations = 1;
  repeated int32 vetVisits = 2;
}

message Dog {
  string name = 1;
  string gender = 2;
  string owner = 3;
  Dog_HealthRecord healthRecord = 4;
}



./src/default.ts

{"rules":[{"rule":"isCool === \"foo\"","value":true},{"rule":"isCool === \"bar\" && breed === \"lab\"","value":true},{"rule":"(isCool === \"bar\") && (breed === \"retriever\")","value":true}],"name":"getPuppyFlag","type":"boolean","default":false}

{"rules":[{"rule":"isCool === \"foo\"","value":"Deadly"},{"rule":"isCool === \"bar\" && breed === \"lab\"","value":"Serious"},{"rule":"(isCool === \"bar\") && (breed === \"retriever\")","value":"Deadly"}],"name":"getPuppyFartRisk","type":"string","default":"Use Caution"}

{"rules":[{"rule":"isCool === \"foo\"","value":"6"}],"name":"getPuppyNumber","type":"number","default":"9"}

{"rules":[],"name":"getGuardDog","type":"Dog","default":" {\n\t\tname: \"Rex\",\n\t\tgender: \"male\",\n\t\thealthRecord: {\n\t\t\tvaccinations: [],\n\t\t\tvetVisits: [new Date(\"2023-12-09\")]\n\t\t}\n\t}"}
```

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "main": "lib/index.js",
   "types": "lib/index.d.ts",
   "bin": {
+    "ts-to-lekko": "bin/ts-to-lekko.ts",
```
Comment (lekko-jonathan) [inline package.json:9]: based on the line below, this probably will need to be .js

File: src/ts-to-lekko.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+const registry = {};
+interfaceNodes.forEach((interfaceNode) => {
+    convertInterfaceToProto(interfaceNode as ts.InterfaceDeclaration, typeChecker, registry);
+})
+
+let protofile = "";
+
+const keys = Object.keys(registry);
+keys.sort((a, b) => b.length - a.length);
```
Comment (lekko-jonathan) [inline src/ts-to-lekko.ts:217]: this.. uh.. well obviously this fails for many dependency graph cases

**Comment by konradjniemiec** [comment]
this is great. not the ugliest code of all time I promise you

**Pull request closed without merging**